packages/gafl-webapp-service/src/handlers/analytics-handler.js

```diff
@@ -47,5 +47,12 @@ export default async (request, h) => {
       [ANALYTICS.seenMessage]: true
     })
   }
-  return h.redirect(addLanguageCodeToUri(request, '/buy'))
+
+  // https://localhost:3043/buy/name
+  const origin = request.headers.origin
+  const referer = request.headers.referer
+  const redirect = referer.replace(origin, '')
```

Comment:
We need to check that we're redirecting to a valid location - so if the host / port match the host / port of the current page, then redirect to the path after host / port

For example:
Referer is https://localhost:3043/buy/name and the current url is https://localhost:3043/buy/process-analytics-preferences, we can redirect to `/buy/name`

However, if the referer is `http://some.shady.url:9999/who/knows/what`, we should redirect to `/buy`